omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -571,7 +571,8 @@ def warn_if_cred_mismatch(opts)
     #
     def save_credentials_to_config
       credentials.legacy_credentials_hash.each do |service, creds|
-        next if service == "chef-server"
+        # Ignore secrets added by add-ons and the keys
+        next if PrivateChef[service].nil?
```

Comment:
@marcparadise I've discussed and double-checked this with @stevendanna, it's catching a bug we had earlier when you'd do the following:

1. install (a current package of) chef-server, set `insecure_addon_compat true`, reconfigure
2. install a current addon, run its reconfigure
3. run chef-server-ctl reconfigure again

it would then try to ingest _all the secrets_ from veil into `PrivateChef` (to end up writing them to running.json), and it would fail when a service didn't exist with a key in PrivateChef.